org-tanglesync.el

```diff
@@ -460,23 +465,14 @@ Uses `org-tanglesync-watch-files` to generate.")
   "A hook to update current buffer contents in the source org file.
 Takes the current contents of the saved file and sync them back to
 the source org file they are originally tangled to."
-  (when (and org-tanglesync-watch-files org-tanglesync-watch-mode)
-    (let ((tfile buffer-file-name)
-          ;;(org-tanglesync-confmap
-          ;; (org-tanglesync-watch-make-watchlist
-          ;;  org-tanglesync-watch-files))
-          ;; -- this should already be initialised, though the issue is
-          ;;    when do we update it?
-          (contbuff (org-tanglesync-get-filedata-buffer buffer-file-name)))
-      (let ((cfile (org-tanglesync-watch-get-conf-source
-                    tfile
-                    org-tanglesync-confmap)))
-        (when cfile
-          (org-tanglesync-watch-perform-sync tfile cfile contbuff)
-          (kill-buffer contbuff))))))
-
-(add-hook 'after-save-hook #'org-tanglesync-watch-save)
-(add-hook 'org-src-mode-hook #'org-tanglesync-user-edit-buffer)
+  (when org-tanglesync-watch-files
+    (let* ((tfile buffer-file-name)
+           (cfile (org-tanglesync-watch-get-conf-source
+                   tfile
+                   org-tanglesync-confmap)))
+      (when cfile
+        (org-tanglesync-watch-perform-sync
+         tfile cfile (org-tanglesync-get-filedata-buffer buffer-file-name))))))
```

Comment:
so the `org-tanglesync-get-filedata-buffer` actually still copies data to a new buffer `*filedata*` (check the function definition).

The rationale behind it was to have a place where the diff function will always know where to look.

This is not really needed for the watch mode, since we will always be pushing from the external file into the org-buffer.

So here I think we just need to write
```lisp
(org-tanglesync-watch-perform-sync
    tfile cfile buffer-file-name)
```

Reply:
it needs to be a buffer name, not a file name. But that is a really good idea.